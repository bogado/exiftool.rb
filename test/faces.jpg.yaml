--- 
:region_applied_to_dimensions_h: 2336
:aeb_sequence_auto_cancel: 0,-,+/Enabled
:color_components: 3
:sensor_width: 3596
:picture_style: None
:color_temp_flash: 6303
:"profile_description_ml-ko-kr": !binary |
  7Lm066mU6528IFJHQiDtlITroZztjIzsnbw=

:white_balance_blue: 0
:orientation: Horizontal (normal)
:user_comment: ""
:self_timer2: 0
:add_original_decision_data: "On"
:media_white_point: 0.95047 1 1.0891
:canon_image_width: 3504
:thumbnail_length: 12170
:color_tone: Normal
:bulb_duration: 0
:rendering_intent: Perceptual
:af_area_x_positions: 8 -555 571 -931 8 947 -555 571 8
:canon_image_type: Canon EOS 20D
:"profile_description_ml-zh-cn": !binary |
  55u45py6IFJHQiDmj4/ov7Dmlofku7Y=

:short_focal: 17 mm
:color_temp_tungsten: 3203
:color_temp_shade: 7000
:region_applied_to_dimensions_w: 3504
:measured_ev: 14.12
:drive_mode: Single-frame Shooting
:saturation: Normal
:flash_exposure_comp: !ruby/object:Rational 
  denominator: 3
  numerator: 2
:af_point_selection_method: Normal
:toning_effect: None
:iso: 400
:profile_date_time: 2003-07-01 00:00:00 -07:00
:scale_factor35efl: 1.6
:macro_mode: Normal
:shutter_curtain_sync: 1st-curtain sync
:af_area_height: 78
:"profile_description_ml-zh-tw": !binary |
  5pW45L2N55u45qmfIFJHQiDoibLlvanmj4/ov7A=

:superimposed_display: "On"
:focal_type: Zoom
:auto_rotate: None
:software: Picasa
:af_area_y_positions: 504 270 270 4 4 4 -262 -262 -496
:focus_mode: One-shot AF
:wb_rggb_levels_shade: 2215 1023 1023 1341
:sharpness_frequency: n/a
:profile_class: Input Device Profile
:lens_type: Unknown (-1)
:green_matrix_column: 0.35332 0.67441 0.09042
:focal_units: 1/mm
:measured_rggb: 641 1024 1024 660
:quality: Fine
:color_temp_auto: 4599
:file_permissions: rw-r--r--
:cmm_flags: Not Embedded, Independent
:source_file: test/faces.jpg
:num_af_points: 9
:bracket_value: 0
:color_temperature: 4300
:exif_version: "0221"
:long_focal: 40 mm
:circle_of_confusion: 0.019 mm
:focal_plane_x_resolution: 3959.322034
:iptc_digest: b443520a10119da99c2550175e6d0efb
:aperture: 4.0
:profile_description_ml: Camera RGB Profile
:manual_flash_output: n/a
:make: Canon
:flash: Off, Did not fire
:slow_shutter: None
:black_mask_bottom_border: 0
:envelope_record_version: 4
:light_value: 14.0
:exposure_mode: Auto
:long_exposure_noise_reduction: "On"
:base_iso: 400
:color_space_data: "RGB "
:focal_length: 17.0 mm
:contrast: Normal
:shutter_speed_value: !ruby/object:Rational 
  denominator: 4000
  numerator: 1
:menu_button_display_position: Previous (top if power off)
:warning: Invalid original decision data
:focal_plane_y_resolution: 3959.322034
:aeb_bracket_value: 0
:focus_continuous: Single
:profile_cmm_type: appl
:measured_ev2: 14.125
:thumbnail_image_valid_area: 0 159 7 112
:aperture_value: 4.0
:continuous_drive: Single
:color_space: sRGB
:mime_type: image/jpeg
:profile_connection_space: "XYZ "
:modify_date: 2012-02-15 22:58:07 -08:00
:blue_matrix_column: 0.15662 0.08336 0.71953
:interop_version: "0100"
:custom_rendered: Normal
:canon_flash_mode: "Off"
:wb_rggb_levels_daylight: 1933 1023 1023 1607
:region_type: 
- Face
- Face
- Face
- Face
- Face
:image_height: 2336
:bracket_shot_number: 0
:y_cb_cr_positioning: Co-sited
:lens_id: Unknown 17-40mm
:red_trc: (Binary data 14 bytes)
:exif_image_width: 3504
:shooting_mode: Depth-of-field AE
:sensor_blue_level: 0
:x_resolution: 72
:xmp_toolkit: XMP Core 5.1.2
:canon_firmware_version: Firmware 2.0.3
:chromatic_adaptation: 1.04788 0.02292 -0.0502 0.02957 0.99049 -0.01706 -0.00923 0.01508 0.75165
:sensor_top_border: 19
:focal_plane_y_size: 15.37 mm
:"shutter-ae_lock": AF/AE lock
:exposure_compensation: 0
:sensor_right_border: 3587
:hyperfocal_distance: 3.76 m
:image_size: 3504x2336
:lens: 17.0 - 40.0 mm
:coded_character_set: UTF8
:black_mask_right_border: 0
:file_name: faces.jpg
:primary_platform: Apple Computer Inc.
:focus_range: Not Known
:scene_capture_type: Standard
:thumbnail_image: (Binary data 12170 bytes)
:vrd_offset: 0
:self_timer: "Off"
:shutter_speed: !ruby/object:Rational 
  denominator: 4000
  numerator: 1
:connection_space_illuminant: 0.9642 1 0.82491
:fov: 68.3 deg
:canon_image_height: 2336
:region_area_x: 
- 0.549658
- 0.858162
- 0.730736
- 0.453196
- 0.118579
:thumbnail_offset: 4036
:wb_bracket_value_ab: 0
:green_trc: (Binary data 14 bytes)
:bracket_mode: "Off"
:interop_index: R98 - DCF basic file (sRGB)
:flash_guide_number: 0
:wb_rggb_levels_fluorescent: 1743 1023 1023 2205
:region_area_w: 
- 0.0667808
- 0.103881
- 0.0687785
- 0.0753424
- 0.11387
:wb_rggb_levels_custom1: 1096 1023 1023 2793
:red_balance: 1.795074
:components_configuration: Y, Cb, Cr, -
:region_area_unit: 
- normalized
- normalized
- normalized
- normalized
- normalized
:digital_gain: 0
:profile_copyright: Copyright 2003 Apple Computer Inc., all rights reserved.
:af_image_height: 2336
:wb_rggb_levels_custom2: 1734 1023 1023 1901
:create_date: 2005-11-26 14:20:34 -08:00
:resolution_unit: inches
:"profile_description_ml-sv-se": "RGB-profil f\xC3\xB6r Kamera"
:target_exposure_time: !ruby/object:Rational 
  denominator: 4008
  numerator: 1
:focal_plane_x_size: 23.04 mm
:serial_number_format: Format 2
:file_number: 326-2647
:region_name: 
- Matthew McEachen
- Karen McEachen
- Ruth McEachen
- James McEachen
- Jamie McEachen
:sequence_number: 0
:device_manufacturer: appl
:wb_rggb_levels_auto: 1822 1023 1023 1709
:digital_zoom: None
:filter_effect: None
:region_applied_to_dimensions_unit: pixel
:sharpness: 0
:flash_firing: Fires
:color_temp_fluorescent: 3961
:exposure_time: !ruby/object:Rational 
  denominator: 4000
  numerator: 1
:color_temp_cloudy: 6000
:region_area_y: 
- 0.248716
- 0.486301
- 0.243365
- 0.360873
- 0.466182
:auto_iso: 100
:wb_rggb_levels_cloudy: 2087 1023 1023 1459
:easy_mode: Manual
:exposure_program: Not Defined
:"profile_description_ml-pt-br": "Perfil RGB de C\xC3\xA2mera"
:directory: test
:control_mode: Camera Local Control
:profile_creator: appl
:exif_tool_version: 8.59
:sensor_height: 2360
:y_cb_cr_sub_sampling: YCbCr4:2:2 (2 1)
:color_temp_daylight: 5200
:"profile_description_ml-fi-fi": Kameran RGB-profiili
:profile_file_signature: acsp
:ettlii: Evaluative
:wb_shift_ab: 0
:y_resolution: 72
:model: Canon EOS 20D
:zoom_source_width: 0
:flash_bits: (none)
:application_record_version: 4
:zoom_target_width: 0
:wb_rggb_levels_flash: 2112 1023 1023 1414
:record_mode: JPEG
:flash_sync_speed_av: Auto
:metering_mode: Center-weighted average
:safety_shift_in_av_or_tv: Enable
:"profile_description_ml-no-no": RGB-kameraprofil
:camera_type: EOS Mid-range
:f_number: 4.0
:serial_number: 0720413239
:flash_activity: 0
:auto_exposure_bracketing: "Off"
:target_aperture: 4
:device_model: ""
:image_width: 3504
:file_type: JPEG
:"profile_description_ml-de-de": "RGB-Profil f\xC3\xBCr Kameras"
:af_image_width: 3504
:red_matrix_column: 0.45427 0.24263 0.01482
:valid_af_points: 9
:af_assist_beam: Emits
:exposure_level_increments: !ruby/object:Rational 
  denominator: 3
  numerator: 1
:file_modify_date: 2012-02-15 22:58:07 -08:00
:bits_per_sample: 8
:tone_curve: Standard
:exif_image_height: 2336
:blue_trc: (Binary data 14 bytes)
:encoding_process: Baseline DCT, Huffman coding
:focal_length35efl: "17.0 mm (35 mm equivalent: 26.6 mm)"
:black_mask_top_border: 0
:set_function_when_shooting: Change quality
:profile_description: Camera RGB Profile
:sensor_red_level: 0
:lens35efl: "17.0 - 40.0 mm (35 mm equivalent: 26.6 - 62.5 mm)"
:black_mask_left_border: 0
:"profile_description_ml-nl-nl": RGB-profiel Camera
:sensor_bottom_border: 2354
:flashpix_version: "0100"
:profile_id: 0
:exif_byte_order: Little-endian (Intel, II)
:wb_bracket_mode: "Off"
:"profile_description_ml-fr-fu": "Profil RVB de l\xE2\x80\x99appareil-photo"
:lens_af_stop_button: ONE SHOT <-> AI SERVO
:photo_effect: "Off"
:mirror_lockup: Disable
:nd_filter: n/a
:"profile_description_ml-ja-jp": !binary |
  44Kr44Oh44OpIFJHQiDjg5fjg63jg5XjgqHjgqTjg6s=

:region_area_h: 
- 0.119007
- 0.1875
- 0.12286
- 0.101884
- 0.189212
:wb_rggb_levels_tungsten: 1382 1023 1023 2574
:file_size: 3.8 MB
:current_iptc_digest: b443520a10119da99c2550175e6d0efb
:canon_exposure_mode: Depth-of-field AE
:blue_balance: 1.683744
:"profile_description_ml-it-it": Profilo RGB Fotocamera
:color_temp_as_shot: 4599
:long_exposure_noise_reduction2: "Off"
:owner_name: unknown
:canon_image_size: Large
:wb_rggb_levels: 1822 1015 1015 1709
:device_attributes: Reflective, Glossy, Positive, Color
:original_decision_data_offset: 3968887
:white_balance_red: 0
:af_area_width: 78
:sensor_left_border: 84
:focal_plane_resolution_unit: inches
:compression: JPEG (old-style)
:"profile_description_ml-es-es": "Perfil RGB para C\xC3\xA1mara"
:optical_zoom_code: n/a
:color_temp_custom1: 2694
:iso_expansion: "Off"
:white_balance: Auto
:"profile_description_ml-da-dk": RGB-beskrivelse til Kamera
:profile_version: 2.2.0
:wb_rggb_levels_as_shot: 1822 1015 1015 1709
:color_temp_custom2: 4301
:date_time_original: 2005-11-26 14:20:34 -08:00
:wb_bracket_value_gm: 0
:canon_model_id: EOS 20D
:af_points_in_focus: 0,1,2,3,4,5,6,7,8
:wb_shift_gm: 0
