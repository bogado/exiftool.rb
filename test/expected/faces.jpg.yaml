---
:file_name: faces.jpg
:file_size: 3966823
:file_type: JPEG
:file_type_extension: JPG
:mime_type: image/jpeg
:exif_byte_order: II
:make: Canon
:model: Canon EOS 20D
:orientation: 1
:x_resolution: 72
:y_resolution: 72
:resolution_unit: 2
:software: Picasa
:modify_date: 2012:02:15 22:58:07
:modify_date_civil: 2012-02-15
:y_cb_cr_positioning: 2
:exposure_time: 0.00025
:f_number: 4
:exposure_program: 0
:iso: 400
:exif_version: '0221'
:date_time_original: 2005:11:26 14:20:34
:date_time_original_civil: 2005-11-26
:create_date: 2005:11:26 14:20:34
:create_date_civil: 2005-11-26
:components_configuration: 1 2 3 0
:shutter_speed_value: '0.000249999045999457'
:aperture_value: 4
:flash: 16
:focal_length: 17
:macro_mode: 2
:self_timer: 0
:quality: 3
:canon_flash_mode: 0
:continuous_drive: 0
:focus_mode: 0
:record_mode: 1
:canon_image_size: 0
:easy_mode: 1
:digital_zoom: 0
:contrast: 0
:saturation: 0
:sharpness: 0
:metering_mode: 5
:focus_range: 2
:canon_exposure_mode: 5
:lens_type: -1
:focal_units: 1
:flash_activity: 0
:flash_bits: 0
:focus_continuous: 0
:zoom_source_width: 0
:zoom_target_width: 0
:photo_effect: 0
:manual_flash_output: 32767
:focal_type: 2
:focal_plane_x_size: 23.0378
:focal_plane_y_size: 15.367
:auto_iso: 100
:base_iso: 400
:measured_ev: 14.125
:target_aperture: 4
:target_exposure_time: '0.000249486608558134'
:exposure_compensation: 0
:white_balance: 0
:slow_shutter: 3
:sequence_number: 0
:optical_zoom_code: 8
:flash_guide_number: 0
:flash_exposure_comp: 0.666666666666667
:auto_exposure_bracketing: 0
:aeb_bracket_value: 0
:control_mode: 1
:measured_ev2: 14.125
:bulb_duration: 0
:camera_type: 252
:auto_rotate: 0
:nd_filter: -1
:self_timer2: 0
:canon_image_type: Canon EOS 20D
:canon_firmware_version: Firmware 2.0.3
:owner_name: unknown
:serial_number: 720413239
:set_function_when_shooting: 1
:long_exposure_noise_reduction: 1
:flash_sync_speed_av: 0
:shutter-ae_lock: 0
:af_assist_beam: 0
:exposure_level_increments: 0
:flash_firing: 0
:iso_expansion: 0
:aeb_sequence_auto_cancel: 0
:superimposed_display: 0
:menu_button_display_position: 0
:mirror_lockup: 0
:af_point_selection_method: 0
:ettlii: 0
:shutter_curtain_sync: 0
:safety_shift_in_av_or_tv: 1
:lens_af_stop_button: 4
:add_original_decision_data: 1
:canon_model_id: 2147484021
:num_af_points: 9
:valid_af_points: 9
:canon_image_width: 3504
:canon_image_height: 2336
:af_image_width: 3504
:af_image_height: 2336
:af_area_width: 78
:af_area_height: 78
:af_area_x_positions: 8 -555 571 -931 8 947 -555 571 8
:af_area_y_positions: 504 270 270 4 4 4 -262 -262 -496
:af_points_in_focus: 511
:thumbnail_image_valid_area: 0 159 7 112
:serial_number_format: 2684354560
:original_decision_data_offset: 3968887
:file_number: 3262647
:bracket_mode: 0
:bracket_value: 0
:bracket_shot_number: 0
:long_exposure_noise_reduction2: 0
:wb_bracket_mode: 0
:wb_bracket_value_ab: 0
:wb_bracket_value_gm: 0
:filter_effect: 0
:toning_effect: 0
:tone_curve: 0
:sharpness_frequency: 0
:sensor_red_level: 0
:sensor_blue_level: 0
:white_balance_red: 0
:white_balance_blue: 0
:color_temperature: 4300
:picture_style: 0
:digital_gain: 0
:wb_shift_ab: 0
:wb_shift_gm: 0
:measured_rggb: 641 1024 1024 660
:vrd_offset: 0
:sensor_width: 3596
:sensor_height: 2360
:sensor_left_border: 84
:sensor_top_border: 19
:sensor_right_border: 3587
:sensor_bottom_border: 2354
:black_mask_left_border: 0
:black_mask_top_border: 0
:black_mask_right_border: 0
:black_mask_bottom_border: 0
:wb_rggb_levels_as_shot: 1822 1015 1015 1709
:color_temp_as_shot: 4599
:wb_rggb_levels_auto: 1822 1023 1023 1709
:color_temp_auto: 4599
:wb_rggb_levels_daylight: 1933 1023 1023 1607
:color_temp_daylight: 5200
:wb_rggb_levels_shade: 2215 1023 1023 1341
:color_temp_shade: 7000
:wb_rggb_levels_cloudy: 2087 1023 1023 1459
:color_temp_cloudy: 6000
:wb_rggb_levels_tungsten: 1382 1023 1023 2574
:color_temp_tungsten: 3203
:wb_rggb_levels_fluorescent: 1743 1023 1023 2205
:color_temp_fluorescent: 3961
:wb_rggb_levels_flash: 2112 1023 1023 1414
:color_temp_flash: 6303
:wb_rggb_levels_custom1: 1096 1023 1023 2793
:color_temp_custom1: 2694
:wb_rggb_levels_custom2: 1734 1023 1023 1901
:color_temp_custom2: 4301
:color_tone: 0
:user_comment: ''
:flashpix_version: '0100'
:color_space: 1
:exif_image_width: 3504
:exif_image_height: 2336
:interop_index: R98
:interop_version: '0100'
:focal_plane_x_resolution: 3959.322034
:focal_plane_y_resolution: 3959.322034
:focal_plane_resolution_unit: 2
:custom_rendered: 0
:exposure_mode: 0
:scene_capture_type: 0
:compression: 6
:thumbnail_offset: 4036
:thumbnail_length: 12170
:xmp_toolkit: XMP Core 5.1.2
:region_applied_to_dimensions_w: 3504
:region_applied_to_dimensions_h: 2336
:region_applied_to_dimensions_unit: pixel
:region_name:
- Matthew McEachen
- Karen McEachen
- Ruth McEachen
- James McEachen
- Jamie McEachen
:region_type:
- Face
- Face
- Face
- Face
- Face
:region_area_x:
- 0.549658
- 0.858162
- 0.730736
- 0.453196
- 0.118579
:region_area_y:
- 0.248716
- 0.486301
- 0.243365
- 0.360873
- 0.466182
:region_area_w:
- 0.0667808
- 0.103881
- 0.0687785
- 0.0753424
- 0.11387
:region_area_h:
- 0.119007
- 0.1875
- 0.12286
- 0.101884
- 0.189212
:region_area_unit:
- normalized
- normalized
- normalized
- normalized
- normalized
:profile_cmm_type: appl
:profile_version: 544
:profile_class: scnr
:color_space_data: 'RGB '
:profile_connection_space: 'XYZ '
:profile_date_time: 2003:07:01 00:00:00
:profile_date_time_civil: 2003-07-01
:profile_file_signature: acsp
:primary_platform: APPL
:cmm_flags: 0
:device_manufacturer: appl
:device_model: ''
:device_attributes: 0 0
:rendering_intent: 0
:connection_space_illuminant: 0.9642 1 0.82491
:profile_creator: appl
:profile_id: 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
:red_matrix_column: 0.45427 0.24263 0.01482
:green_matrix_column: 0.35332 0.67441 0.09042
:blue_matrix_column: 0.15662 0.08336 0.71953
:media_white_point: 0.95047 1 1.0891
:chromatic_adaptation: 1.04788 0.02292 -0.0502 0.02957 0.99049 -0.01706 -0.00923 0.01508
  0.75165
:red_trc: "(Binary data 14 bytes, use -b option to extract)"
:green_trc: "(Binary data 14 bytes, use -b option to extract)"
:blue_trc: "(Binary data 14 bytes, use -b option to extract)"
:profile_description: Camera RGB Profile
:profile_copyright: Copyright 2003 Apple Computer Inc., all rights reserved.
:profile_description_ml: Camera RGB Profile
:profile_description_ml-es-es: Perfil RGB para Cámara
:profile_description_ml-da-dk: RGB-beskrivelse til Kamera
:profile_description_ml-de-de: RGB-Profil für Kameras
:profile_description_ml-fi-fi: Kameran RGB-profiili
:profile_description_ml-fr-fu: Profil RVB de l’appareil-photo
:profile_description_ml-it-it: Profilo RGB Fotocamera
:profile_description_ml-nl-nl: RGB-profiel Camera
:profile_description_ml-no-no: RGB-kameraprofil
:profile_description_ml-pt-br: Perfil RGB de Câmera
:profile_description_ml-sv-se: RGB-profil för Kamera
:profile_description_ml-ja-jp: カメラ RGB プロファイル
:profile_description_ml-ko-kr: 카메라 RGB 프로파일
:profile_description_ml-zh-tw: 數位相機 RGB 色彩描述
:profile_description_ml-zh-cn: 相机 RGB 描述文件
:current_iptc_digest: b443520a10119da99c2550175e6d0efb
:envelope_record_version: 4
:coded_character_set: !binary |-
  GyVH
:application_record_version: 4
:iptc_digest: b443520a10119da99c2550175e6d0efb
:image_width: 3504
:image_height: 2336
:encoding_process: 0
:bits_per_sample: 8
:color_components: 3
:y_cb_cr_sub_sampling: 2 1
:aperture: 4
:drive_mode: 2
:image_size: 3504x2336
:lens: 17
:lens_id: -1
:megapixels: 8.185344
:warning: Invalid original decision data
:shooting_mode: 5
:shutter_speed: 0.00025
:wb_rggb_levels: 1822 1015 1015 1709
:blue_balance: 1.68374384236453
:fov: 66.9411904372194
:hyperfocal_distance: 3.85099482285285
:light_value: 13.9657842846621
:red_balance: 1.79507389162562
